### ISSUE TITLE
Several questions on the paper

### ISSUE THREAD

**Issue by alephpi** [HumeAI/tada#13]
Hi, thank you for open source this work. I have several questions on the paper.
 
1. The paper doesn't mention exact number of the lookahead shift K. I guess it's set to 5 for both training and inference?https://github.com/HumeAI/tada/blob/91559120256fff3ff4e06042af061c3aa20e5099/tada/modules/tada.py#L50
Would you justify the choice? Does that mean the model would try to predict 5 consecutive `<eos>` on the text stream at the end of a sentence, to wait the audio stream finishes?
2. Which dataset do you evaluate on for the results in Table 5 and 6?
3. In Section 4.2 you said 
    > We train the model with the flow matching loss $$\mathcal{L}_{flow}$$ and two optional losses: text cross-entropy loss $$\mathcal{L}_{CE}$$ and text knowledge distillation loss $$\mathcal{L}_{KD}$$.

    And in In Section 5.3 you said

    > Throughout both phases, we apply a cross-entropy loss (weight $$λ_{CE} = 0.05$$ ) to adapt the model to the nuances of spoken text, alongside a KL divergence loss (weight $$λ_{KD} = 0.05$$ ) against their base Llama model to preserve prior knowledge.

    I understand that TADA is not only made for TTS but for more general SLM. However, if we focus on the TTS task, do we need to predict the text and use the text losses? I notice the ablation study in Table 6, seems there is a slight performance degradation on CER (possibly due to modality gap when trying to make the model focus more on text prediction task), I just wonder if you have done a ablation study that totally removes these two losses and report the CER. And shouldn't there be a row in Table 6 with TADA-1B ($$λ_{CE} = 0.05,λ_{KD} = 0.05$$), as this is the standard setting?

**Comment by trungd** [comment]
Hi @alephpi,

Thanks for your interest in the paper. To answer your questions:

1. Yes acoustic features are shifted 5 positions, which means our TTS has 5 text token lookahead. The model keeps generating more text until the target number of acoustic tokens are reached (so technically not eos but some continuation). 5 should offer enough forward context while keeping the latency low.

2. It's the voice cloning setup in Table 2 with PPL as in Table 4.

3. Yes we tried removing those two losses but it didn't significantly improve the TTS performance. Ablations in Table 6 were run at a small scale before the main final model so we don't have the base number. However the conclusion holds that both CE and KD losses are useful at a negligible cost to the TTS performance.

Hope these answer your questions! We will review the paper and add details if missing.

**Comment by alephpi** [comment]
Thank you for quick response. 
> so technically not eos but some continuation

That made me wonder, if it's continuation, how do you make the text-audio training pairs? I guess you use some mask tokens at the first K position in audio stream, but what exactly for the last K positions in text stream if not using eos or pad or any other special tokens. If it's literally 'continuation', does that mean you first put text stream to the base llama model to generate K more tokens and join it to the training data? Or maybe you just don't control to end the text stream generation, the generation is ended when the step reaches L+K, and during the training the last K text tokens are just removed from loss calculation?

> Yes we tried removing those two losses but it didn't significantly improve the TTS performance

So even for TTS task, TADA is predicting text during training. How about the inference, is TADA still predicting text under the hood? (I know I could check the code but let me take advantage of you :))

**Comment by trungd** [comment]
The model should predict text in every step. In the TTS mode, we simply don't use the predicted text but force feed the text 1 by 1 from the target sentence. Hope that makes sense!
